packages/ui/src/views/connectors/connectors-list/connectors-list-page.tsx

```diff
@@ -0,0 +1,91 @@
+import { FC } from 'react'
+
+import { Button, ListActions, NoData, Pagination, SearchBox, Spacer } from '@/components'
+import { useRouterContext } from '@/context'
+import { useDebounceSearch } from '@/hooks'
+import { SandboxLayout } from '@/views'
+import { cn } from '@utils/cn'
+
+import { ConnectorsList } from './connectors-list'
+import { ConnectorListPageProps } from './types'
+
+const ConnectorListPage: FC<ConnectorListPageProps> = ({
+  searchQuery,
+  setSearchQuery,
+  isError,
+  errorMessage,
+  useTranslationStore,
+  currentPage,
+  totalPages,
+  goToPage,
+  isLoading,
+  connectors,
+  ...props
+}) => {
+  const { t } = useTranslationStore()
+  const { navigate } = useRouterContext()
+
+  const { search: searchInput, handleSearchChange: handleInputChange } = useDebounceSearch({
+    handleChangeSearchValue: (val: string) => setSearchQuery(val.length ? val : undefined),
+    searchValue: searchQuery || ''
+  })
+
+  if (isError) {
+    return (
+      <NoData
+        textWrapperClassName="max-w-[350px]"
+        iconName="no-data-error"
+        title={t('views:noData.errorApiTitle', 'Failed to load', {
+          type: 'connectors'
+        })}
+        description={[
+          errorMessage ||
+            t(
+              'views:noData.errorApiDescription',
+              'An error occurred while loading the data. Please try again and reload the page.'
+            )
+        ]}
+        primaryButton={{
+          label: t('views:notFound.button', 'Reload page'),
+          onClick: () => {
+            navigate(0) // Reload the page
```

Comment:
does having the 0 here , reload page?

Reply:
Yes, `navigate(0)` is reload, `navigate(1)` is forward one step in the browser's history and `navigate(-1)` is back.